File: Java/Class-2.1/src/org/launchcode/main/Cat.java
package org.launchcode.main;

public class Cat implements Cloneable {

	String name;

	public Cat() {
		this.name = "cat";
	}

	public Cat(String name) {
		this.name = name;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public static void makeNoise() {
		System.out.println("ROAR!");
	}

	public Cat clone() throws CloneNotSupportedException {
		return (Cat) super.clone();
	}

}

File: CSharp/LC101-Unit2/Class-2.1/Cat.cs
using System;

namespace Class_2._1
{
    public class Cat
    {
        String name;


        public Cat()
        {
            name = "cat";
        }

        public Cat(String name)
        {
            this.name = name;
        }

        public String getName()
        {
            return name;
        }

        public void setName(String name)
        {
            this.name = name;
        }

        // 2.4.2 Static Methods
        public static void makeNoise(String noise)
        {
            if(noise == null) {
                Console.WriteLine("Roar!");
            }
            else
            {
                Console.WriteLine(noise);
            }
        }
                
    }
}
